### PR TITLE
[FLINK-2386] Add new KafkaConsumer, based on Kafka's Simple Consumer API

### PR DESCRIPTION
**Pull request by rmetzger**
This pull request is based on https://github.com/apache/flink/pull/1028.
I'm creating another pull request because I rebased everything to the current master.

Since there were no -1 votes #1028, I assume that this PR is close to be mergeable.

### CODE REVIEW

**Comment by StephanEwen** [comment]
This needs to go into 0.9.1, so the diamond operators are going to be a problem

There is no difference, currently, between the 0.8.2 and the 0.8.1 Kafka consumers. Should we even differentiate between them, or only port the `PersistentKafkaSource` to extend the `FlinkKafkaConsumer`

**Comment by rmetzger** [comment]
Damn. You are right. I'll add the full type parameters again for the 0.9.1 backport (once this one is merged).

> Should we even differentiate between them, or only port the PersistentKafkaSource to extend the FlinkKafkaConsumer

That's a good question. I though that having the two user-facing classes would allow us to "silently" change the implementation for the different Kafka versions if needed (for example when the new consumer API is out, and we know its fully backwards compatible).
But I guess this is a matter of taste. I can change the name to FlinkKafkaConsumer (without the version suffix) if you want.
Do you think having two `PersistentKafkaSource` in different packages is a good idea? 
I don't know whether Sink / Source or Producer / Consumer are better pairs of names.

**Comment by StephanEwen** [comment]
Concerning the `FlinkKafkaConsumer081` and `FlinkKafkaConsumer082` - lets keep them exactly to fix bugs underneath the hood.

I would actually move all this from the `flink-connector-kafka-083`to the other kafka connector project and replace the old `PersistentKafkaSource`.

In a sepaate patch, we should we add some convenience constructors, where one can directly pass the list of brokers, rather than a properties object.

**Comment by rmetzger** [comment]
Okay, will do.

**Comment by rmetzger** [comment]
Okay, I've relocated the source into the `flink-connector-kafka` module.
I added a convenience class `PersistentKafkaSource` in the exact same package as in the old module, with the same Ctor arguments. So 0.9.0 users can just set their version to 0.9.1 without touching the code.

File: flink-contrib/flink-storm-compatibility/flink-storm-compatibility-core/src/main/java/org/apache/flink/stormcompatibility/api/FlinkLocalCluster.java [hunk 1]
```diff
@@ -41,8 +41,7 @@ public void submitTopology(final String topologyName, final Map<?, ?> conf, fina
 
 	public void submitTopologyWithOpts(final String topologyName, final Map<?, ?> conf, final FlinkTopology topology,
 			final SubmitOptions submitOpts) throws Exception {
-		ClusterUtil
-				.startOnMiniCluster(topology.getStreamGraph().getJobGraph(topologyName), topology.getNumberOfTasks());
+		ClusterUtil.startOnMiniCluster(topology.getStreamGraph().getJobGraph(topologyName), topology.getNumberOfTasks(), -1);
```
Comment (hsaputra) [inline flink-contrib/flink-storm-compatibility/flink-storm-compatibility-core/src/main/java/org/apache/flink/stormcompatibility/api/FlinkLocalCluster.java]: Why is this changed? I think by default will also set it to -1

File: flink-core/src/main/java/org/apache/flink/util/NetUtils.java [hunk 2]
```diff
@@ -48,8 +48,9 @@ public static String getHostnameFromFQDN(String fqdn) {
 	 * Works also for ipv6.
 	 *
 	 * See: http://stackoverflow.com/questions/2345063/java-common-way-to-validate-and-convert-hostport-to-inetsocketaddress
+	 * @return URL object for accessing host and Port
 	 */
-	public static void ensureCorrectHostnamePort(String hostPort) {
+	public static URL ensureCorrectHostnamePort(String hostPort) {
```
Comment (hsaputra) [inline flink-core/src/main/java/org/apache/flink/util/NetUtils.java]: Since it is now returning something, could we change the name to reflect it properly, like ``generateCorrectUrlFromHostnameAndPort``

**Comment by rmetzger** [comment]
I addressed @hsaputra's comments & rebased to the latest master.

**Comment by StephanEwen** [comment]
Merging this...
